### ISSUE TITLE
[PERFORMANCE] Not acceptable loading times in a sequence of actions 

### ISSUE THREAD

**Issue by markusmann-vg** [hzi-braunschweig/SORMAS-Project#8611]
### Feature Description
We have to ensure that the use cases that are processed a hundred times in a health departmentent are doable without not acceptable loading times.  

### Problem Description
The following use case results in loading times that cannot be accepted and actually block the health departments from a productive way of working:
Case Edit -> Create new task -> Search in records -> Save -> Open events & create documents.
The Query Logs are available [here](https://wiki.vitagroup.ag/display/SORMAS/Performance+Taskforce+2022): 

![Bildschirmfoto 2022-03-29 um 15 33 47](https://user-images.githubusercontent.com/62540178/160623299-43bbc2fa-c280-4835-8928-15f958847b1a.png)

### Additional Information
The Internet connection in general is consistently stable.

**Comment by FredrikSchaeferVitagroup** [comment]
> Since the overall description is a bit vague I had a look at the logs. They only provide two points we can work on:
> 
> 1. There is a TransactionRolledbackException related to TaskResource.postTasks, so it looks like some external system is posting tasks to SORMAS. Would be good to know what is connected there. Maybe this is badly affecting the overall performance of the system

I just talked to a Climedo developer. They have modules working with tasks. 
One module seems to create a task every time a Contact, Case, or Person gets updated. 
Another module also uses tasks. It is about automation of sending quarantine orders (Quarantänebescheide). A user in SORMAS can create a specific task and assign it to the climedo user. Several times a day, the climedo user goes through all assigned tasks and automatically sends quarantine orders for each. Once this is done, Climedo uses to query to update all related tasks to `DONE`.
However, those modules are not active for all instances. Furthermore, the instance reporting the performance trouble here is not connected to Climedo at all.

It may very well be that the department uses other software alongside SORMAS, for example C.I.S.S., [which probably also uses tasks](https://cov-it.de/module).
> 2. There are multiple long queries related to external share infos:
> 
> > select case2_.uuid as col_0_0_, (select count(externalsh3_.id) from externalshareinfo e
> > xternalsh3_ cross join cases case4_ where externalsh3_.caze_id=case4_.id and externalsh3_.caze_id=case1_.id group by externalsh3_.caze_id) as col_1_0_, externalsh0_.creationDate as col_2_0_,
> > externalsh0_.status as col_3_0_ from externalshareinfo externalsh0_ left outer join cases case1_ on externalsh0_.caze_id=case1_.id left outer join cases case2_ on externalsh0_.caze_id=case2
> > _.id where (concat(case1_.id, externalsh0_.creationDate) in (select concat(externalsh5_.caze_id, max(externalsh5_.creationDate)) from externalshareinfo externalsh5_ where externalsh5_.caze_i
> 
> In the context of this issue I'd suggest to identify where this long query is coming from and fixing.

The externalshareinfo is used in the context of S2S. Should this ticket be moved to the interactions backlog?

**Comment by Marko-ilmkreis** [comment]
@FredrikSchaeferVitagroup I can confirm that CISS also use tasks to get the work, i.e. case investigation, contact investigation and so on.

**Comment by MartinWahnschaffe** [comment]
I'd close this, as it's too vague.
